ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/MigrationConfig.java

```diff
@@ -111,36 +96,29 @@ private MigrationConfig(final Map<String, String> configs, final String id) {
         ), configs);
   }
 
-  @SuppressFBWarnings("DM_EXIT")
-  private static String getServiceId(final Map<String, String> configs) {
+  private static String getServiceId(final Map<String, String> configs) throws MigrationException {
     final String ksqlServerUrl = configs.get(MigrationConfig.KSQL_SERVER_URL);
     if (ksqlServerUrl == null) {
-      LOGGER.error("Missing required property: " + MigrationConfig.KSQL_SERVER_URL);
-      System.exit(1);
-    }
-
-    final KsqlRestClient client;
-    try {
-      client = KsqlRestClient.create(
-          ksqlServerUrl,
-          Collections.EMPTY_MAP,
-          Collections.EMPTY_MAP,
-          Optional.empty()
-      );
-    } catch (KsqlRestClientException e) {
-      LOGGER.error("Invalid ksql server URL: " + ksqlServerUrl);
-      System.exit(1);
-      return null;
+      throw new MigrationException("Missing required property: " + MigrationConfig.KSQL_SERVER_URL);
     }
 
-    final RestResponse<ServerInfo> response = client.getServerInfo();
-    client.close();
+    final Client client = MigrationsUtil.getKsqlClient(ksqlServerUrl);
+    CompletableFuture<ServerInfo> response = client.serverInfo();
 
-    if (!response.isSuccessful()) {
-      LOGGER.error("Failed to query " + MigrationConfig.KSQL_SERVER_URL + "/info");
-      System.exit(1);
+    try {
+      final String serviceId = response.get().getKsqlServiceId();
+      return serviceId;
+    } catch (InterruptedException e) {
+      throw new MigrationException("Interrupted while attempting to connect to "
+          + ksqlServerUrl + "/info");
+    } catch (ExecutionException e) {
+      if (e.getCause() instanceof IllegalStateException) {
```

Comment:
What results in an IllegalStateException? The error message suggests that's what we get if the server URL is invalid? I'm surprised that'd result in an IllegalStateException.

Reply:
It occurs when the response is in an unexpected format (missing a field etc). That would happen if the url provided also happened to have an /info endpoint, but I'm also starting to realize that it could also happen if the java client and the server have incompatible versions.

Will update the error message

Reply:
We should check up front whether the version is compatible or not, and throw a more specific error message if so. If we do that, hopefully we won't need to check for this specific exception type here anymore (it's weird to check for IllegalStateException since those shouldn't happen under normal operation).